File: JavaSource/src/main/java/ru/akhcheck/patterns/mediator/vehicles/Plane.java
package ru.akhcheck.patterns.mediator.vehicles;

/** Самолёт: просит полосу у диспетчера и садится, если её дали. */
public class Plane extends Vehicle {
    private PlaneCommand status = PlaneCommand.CONNECTED;

    public Plane(String name) {
        super(name);
    }

    public PlaneCommand getStatus() {
        return status;
    }

    /** Просит полосу и садится, только если её дали. */
    public void land() {
        status = PlaneCommand.WAITING;
        System.out.println(getName() + ": requesting the runway");

        boolean allowed = notifyDispatcher("request_landing");
        if (!allowed) {
            System.out.println(getName() + ": runway is busy, staying in the air");
            return;
        }

        status = PlaneCommand.LANDING;
        System.out.println(getName() + ": landing");
        hasLanded();
    }

    public void hasLanded() {
        status = PlaneCommand.LANDED;
        System.out.println(getName() + ": landed");
        notifyDispatcher("runway_free");
    }
}

File: Mediator/python-source/vehicles/Plane.py
from __future__ import annotations

from .PlaneCommands import PlaneCommand
from .Vehicle import Vehicle


class Plane(Vehicle):
    def __init__(self, name: str) -> None:
        super().__init__(name)
        self._status = PlaneCommand.CONNECTED

    def get_status(self) -> PlaneCommand:
        return self._status

    def land(self) -> None:
        """Просит полосу и садится, только если её дали."""
        self._status = PlaneCommand.WAITING
        print(f"{self.name}: requesting the runway")

        allowed = self.notify("request_landing")
        if not allowed:
            print(f"{self.name}: runway is busy, staying in the air")
            return

        self._status = PlaneCommand.LANDING
        print(f"{self.name}: landing")
        self.has_landed()

    def has_landed(self) -> None:
        self._status = PlaneCommand.LANDED
        print(f"{self.name}: landed")
        self.notify("runway_free")
